src/rebar3_hex_file.erl

```diff
@@ -1,35 +1,93 @@
 %% @private
 -module(rebar3_hex_file).
 
--export([ expand_paths/2
-        , update_app_src/2
-        ]).
+-include_lib("kernel/include/file.hrl").
+
+-export([
+    expand_paths/2,
+    update_app_src/2
+]).
 
 expand_paths(Paths, Dir) ->
-    AbsDir = filename:absname(Dir),
-    Files = lists:flatmap(fun dir_files1/1, [filename:join(Dir, P) || P <- Paths]),
-    [{relative_path(F1, AbsDir), F1} || F1 <- filter_regular(Files)].
+    AbsDir = expand(rebar3_hex:to_binary(Dir)),
```

Comment:
As mentioned this is pretty much a 1:1 with the mix hex algo and the path related functions are pretty much a 1:1 with utilities from stdlib in elixir. I think this path makes sense, especially considering I'd like to reach the point to where most of the time we working with binaries vs strings, when possible.